src/Common/src/Interop/Windows/Kernel32/Interop.SetConsoleCtrlHandler.cs

```diff
@@ -16,6 +16,6 @@ internal partial class Kernel32
         internal delegate bool ConsoleCtrlHandlerRoutine(int controlType);
 
         [DllImport(Libraries.Kernel32, SetLastError = true)]
-        internal static extern bool SetConsoleCtrlHandler(ConsoleCtrlHandlerRoutine handler, bool addOrRemove);
+        internal static extern bool SetConsoleCtrlHandler(IntPtr handler, bool addOrRemove);
```

Comment:
Passing the delegate here was resulting in managed allocations?  Of what?  Seems like this change would avoid a marshaling frame, but was that showing up in some profile as a hotspot?

Reply:
Apparently I was hallucinating. Reverting this part. :)